internal/server/deviceserver.go

```diff
@@ -55,3 +89,15 @@ func (s *deviceServer) NotifyEvent(ctx context.Context, in *pb.EventNotification
 	}
 	return &pb.EventReceipt{}, nil
 }
+
+func getTimeout(metadata map[string]string) time.Duration {
+	timeout := 300 * time.Second // Deafult timeout
+	if timeoutStr, found := metadata["ansible-playbook-timeout"]; found {
+		timeoutVal, err := strconv.Atoi(timeoutStr)
+		if err != nil {
```

Comment:
If the timeout provided by the user is invalid, shouldn't it be logged?
On the other hand, we are expected to validate it as part of the API, therefore it makes sense we shouldn't fail here.
And yet, if the intention is to reuse this worker by other clients at some point and not only by the operator, a warning might be useful.

Reply:
Added